source/source.go

```diff
@@ -25,11 +30,34 @@ const (
 	hostnameAnnotationKey = "external-dns.alpha.kubernetes.io/hostname"
 	// The annotation used for defining the desired ingress target
 	targetAnnotationKey = "external-dns.alpha.kubernetes.io/target"
-	// The value of the controller annotation so that we feel resposible
+	// The annotation used for defining the desired DNS record TTL
+	ttlAnnotationKey = "external-dns.alpha.kubernetes.io/ttl"
+	// The value of the controller annotation so that we feel responsible
 	controllerAnnotationValue = "dns-controller"
 )
 
+const (
+	ttlMinimum = 1
+	ttlMaximum = math.MaxUint32
+)
+
 // Source defines the interface Endpoint sources should implement.
 type Source interface {
 	Endpoints() ([]*endpoint.Endpoint, error)
 }
+
+func getTTLFromAnnotations(annotations map[string]string) (endpoint.TTL, error) {
+	ttlNotConfigured := endpoint.TTL(0)
```

Comment:
maybe makes sense to make ttlNotConfigured as a constant in endpoint package, and then reuse in other places to verify if ttl is set or not

Reply:
I like this but it's a nit. Let's do it later.